[Bug]: Config installer component fails silently if malformed

### Is there an existing issue for this?

- [x] I have searched the existing issues

### What happened?

When including a config component in a manifest, it fails silently if the `uninstall` section is missing.

### Steps to reproduce?

1. Create an extension with [Example Config.dnn](https://github.com/user-attachments/files/18855581/Example.Config.dnn.txt) as the manifest
2. Install the extension

### Current Behavior

![Failed install screenshot](https://github.com/user-attachments/assets/14b56925-7e28-4e1b-b5ef-9e1faa97c790)

### Expected Behavior

Either a clear error message or the install continues without issue

### Relevant log output

```shell
2025-02-18 17:33:12,575 [Thread:72][ERROR] DotNetNuke.Services.Exceptions.Exceptions - DotNetNuke.Services.Exceptions.PageLoadException: Object reference not set to an instance of an object. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at DotNetNuke.Services.Installer.Installers.ConfigInstaller.ReadManifest(XPathNavigator manifestNav)
   at DotNetNuke.Services.Installer.Installers.InstallerFactory.GetInstaller(XPathNavigator manifestNav, PackageInfo package)
   at DotNetNuke.Services.Installer.Installers.PackageInstaller.ReadComponents(XPathNavigator manifestNav)
   at DotNetNuke.Services.Installer.Installers.PackageInstaller.ReadManifest(XPathNavigator manifestNav)
   at DotNetNuke.Services.Installer.Installers.PackageInstaller..ctor(String packageManifest, InstallerInfo info)
   at DotNetNuke.Services.Installer.Installer.ProcessPackages(XPathNavigator rootNav)
   at DotNetNuke.Services.Installer.Installer.ReadManifest(Boolean deleteTemp)
   at Dnn.PersonaBar.Extensions.Components.InstallController.GetInstaller(Stream stream, String fileName, Int32 portalId, String legacySkin, Boolean isPortalPackage)
   at Dnn.PersonaBar.Extensions.Components.InstallController.ParsePackage(PortalSettings portalSettings, UserInfo user, String filePath, Stream stream)
   at Dnn.PersonaBar.Extensions.Services.ExtensionsController.<>c__DisplayClass42_1.<UploadFileAction>b__1(Object state)
   at System.Web.Util.SynchronizationHelper.SafeWrapCallback(Action action)
   --- End of inner exception stack trace ---
```

### Anything else?

_No response_

### Affected Versions

9.13.7 (alpha)

### What browsers are you seeing the problem on?

_No response_

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/DNN Platform/Library/Services/Installer/Installers/ConfigInstaller.cs
+++ b/DNN Platform/Library/Services/Installer/Installers/ConfigInstaller.cs
@@ -135,32 +135,55 @@ namespace DotNetNuke.Services.Installer.Installers
             }
         }
 
-        /// <summary>The ReadManifest method reads the manifest file for the config compoent.</summary>
+        /// <summary>The ReadManifest method reads the manifest file for the config component.</summary>
         public override void ReadManifest(XPathNavigator manifestNav)
         {
             this.fileName = Util.ReadAttribute(manifestNav, "fileName");
             this.uninstallFileName = Util.ReadAttribute(manifestNav, "unInstallFileName");
 
-            if (string.IsNullOrEmpty(this.fileName))
+            if (!string.IsNullOrEmpty(this.fileName))
             {
-                XPathNavigator nav = manifestNav.SelectSingleNode("config");
-
-                // Get the name of the target config file to update
-                XPathNavigator nodeNav = nav.SelectSingleNode("configFile");
-                string targetFileName = nodeNav.Value;
-                if (!string.IsNullOrEmpty(targetFileName))
-                {
-                    this.targetFile = new InstallFile(targetFileName, string.Empty, this.Package.InstallerInfo);
-                }
-
-                // Get the Install config changes
-                nodeNav = nav.SelectSingleNode("install");
-                this.installConfig = nodeNav.InnerXml;
-
-                // Get the UnInstall config changes
-                nodeNav = nav.SelectSingleNode("uninstall");
-                this.unInstallConfig = nodeNav.InnerXml;
+                return;
             }
+
+            XPathNavigator nav = manifestNav.SelectSingleNode("config");
+            if (nav is null)
+            {
+                this.Log.AddFailure("config element was missing");
+                return;
+            }
+
+            // Get the name of the target config file to update
+            XPathNavigator nodeNav = nav.SelectSingleNode("configFile");
+            if (nodeNav is null)
+            {
+                this.Log.AddFailure("configFile element was missing");
+                return;
+            }
+
+            string targetFileName = nodeNav.Value;
+            if (!string.IsNullOrEmpty(targetFileName))
+            {
+                this.targetFile = new InstallFile(targetFileName, string.Empty, this.Package.InstallerInfo);
+            }
+
+            // Get the Install config changes
+            nodeNav = nav.SelectSingleNode("install");
+            if (nodeNav is null)
+            {
+                this.Log.AddFailure("install element was missing");
+            }
+
+            this.installConfig = nodeNav?.InnerXml ?? Null.NullString;
+
+            // Get the UnInstall config changes
+            nodeNav = nav.SelectSingleNode("uninstall");
+            if (nodeNav is null)
+            {
+                this.Log.AddWarning("uninstall element was missing");
+            }
+
+            this.unInstallConfig = nodeNav?.InnerXml ?? Null.NullString;
         }
 
         /// <summary>

